src/models/testSuite.ts

```diff
@@ -1,50 +1,42 @@
 import {EntityMap} from '@models/entityMap';
-import {ObjectRef} from '@models/objectRef';
 import {TestSuiteExecution} from '@models/testSuiteExecution';
 
 import {Execution} from './execution';
 import {Variables} from './variable';
 
-export type TestSuiteStepDelay = {
-  duration: number;
-};
-
-export type TestSuiteStepExecuteTest = {
-  namespace?: string;
-  name: string;
-  type: string;
-};
-
-export interface TestSuiteStepAbstract {
-  stopTestOnFailure: boolean;
-  // Used for form
-  id?: string | number;
+export interface AdditionalStepFields extends TestSuiteStep {
+  stopTestOnFailure?: boolean;
+  id?: string;
 }
 
-export interface TestSuiteStepWithExecute extends TestSuiteStepAbstract {
-  execute: TestSuiteStepExecuteTest;
+export interface TestSuiteStep {
+  delay?: string;
+  test?: string;
 }
-
-export interface TestSuiteStepWithDelay extends TestSuiteStepAbstract {
-  delay: TestSuiteStepDelay;
+export interface TestSuiteBatchStep {
+  stopTestOnFailure: boolean;
+  execute: TestSuiteStep[];
 }
 
-export type TestSuiteStep = TestSuiteStepWithExecute | TestSuiteStepWithDelay;
+export interface LocalStep extends AdditionalStepFields {
```

Comment:
It could be declared in `SettingsTests.tsx`, as it's purely local for that component.